kotlin/find-route/src/main/java/com/esri/arcgisruntime/sample/findroute/MainActivity.kt

```diff
@@ -0,0 +1,325 @@
+          // NOTE: to get turn-by-turn directions the route parameters
+          //  must have the isReturnDirections parameter set to true.
+          val directions = route.directionManeuvers
+          setupBottomSheet(directions)
+
+          // when the route is solved, hide the FAB
+          directionFab.visibility = View.GONE
+        } else {
+          val message = "Error loading route parameters."
+          Log.e(TAG, message)
+          Toast.makeText(this, message, Toast.LENGTH_LONG).show()
+        }
+      } catch (e: Exception) {
+        val message = "Error creating route parameters: ${e.message}"
+        Log.e(TAG, message)
+        Toast.makeText(this, message, Toast.LENGTH_LONG).show()
+      }
+    }
+  }
+
+  /**
+   * Set up the source, destination and route symbols.
+   */
+  private fun setupSymbols() {
+    try {
+      val startDrawable =
+        ContextCompat.getDrawable(this, R.drawable.ic_source) as BitmapDrawable?
+
+      val pinSourceSymbol = PictureMarkerSymbol.createAsync(startDrawable).get()
+      pinSourceSymbol.apply {
+        loadAsync()
+        addDoneLoadingListener {
+          // add a new graphic as start point
+          val sourcePoint = Point(
+            -117.15083257944445,
+            32.741123367963446,
+            SpatialReferences.getWgs84()
+          )
+          // add it to the graphics overlay
+          graphicsOverlay.graphics.add(Graphic(sourcePoint, this))
+        }
+        // make the graphic smaller
+        width = 30f
+        height = 30f
+        offsetY = 20f
+      }
+    } catch (e: Exception) {
+      val message = "Error loading picture marker symbol: ${e.message}"
+      Log.e(TAG, message)
+      Toast.makeText(this, message, Toast.LENGTH_LONG).show()
+    }
+    try {
+      val endDrawable =
+        ContextCompat.getDrawable(this, R.drawable.ic_destination) as BitmapDrawable?
+
+      val pinDestinationSymbol = PictureMarkerSymbol.createAsync(endDrawable).get()
+      pinDestinationSymbol.apply {
+        loadAsync()
+        addDoneLoadingListener {
+          // add a new graphic as end point
+          val destinationPoint = Point(
+            -117.15557279683529,
+            32.703360305883045,
+            SpatialReferences.getWgs84()
+          )
+          // add it to the graphics overlay
+          graphicsOverlay.graphics.add(Graphic(destinationPoint, this))
+        }
+        // make the graphic smaller
+        width = 30f
+        height = 30f
+        offsetY = 20f
+      }
+    } catch (e: Exception) {
+      val message = "Error loading picture marker symbol: ${e.message}"
+      Log.e(TAG, message)
+      Toast.makeText(this, message, Toast.LENGTH_LONG).show()
+    }
+  }
+
+  /** Creates a bottom sheet to display a list of direction maneuvers.
+   *
+   * @param directions a list of DirectionManeuver which represents the route
+   */
+  private fun setupBottomSheet(directions: List<DirectionManeuver>) {
```

Comment:
alot of the calls in this function look like they could be conflated into

bottomSheetBehaviour.apply {
...
...
...
}

and 

bottomSheet.apply {
...
}

Reply:
I've made this change but it created a nested apply{apply{}} block. Are we trying to avoid those? I can take it out if needed.

Reply:
I think the directionsListView.apply can be moved out of the bottomSheetBehavious apply block, but maybe that's even more confusing? Up to you!